modules/oap/collector/core/model/exporter.go

```diff
@@ -16,59 +16,99 @@ package model
 
 import (
 	"context"
+	"time"
 
 	"github.com/erda-project/erda-infra/base/logs"
-	"github.com/erda-project/erda/modules/oap/collector/common"
+	"github.com/erda-project/erda/modules/core/monitor/log"
+	"github.com/erda-project/erda/modules/core/monitor/metric"
+	"github.com/erda-project/erda/modules/msp/apm/trace"
 	"github.com/erda-project/erda/modules/oap/collector/core/model/odata"
+	"github.com/erda-project/erda/modules/oap/collector/lib"
 )
 
 type RuntimeExporter struct {
 	Name     string
+	DType    odata.DataType
 	Logger   logs.Logger
 	Exporter Exporter
 	Filter   *DataFilter
-	Timer    *common.RunningTimer
-	Buffer   *odata.Buffer
+
+	Buffer           *odata.Buffer
+	Timer            *time.Timer
+	Interval, Jitter time.Duration
+}
+
+func (re *RuntimeExporter) Add(od odata.ObservableData) {
+	re.Buffer.Push(od)
+
+	if re.Buffer.Full() {
```

Comment:
consider thread safety

Reply:
There is a mutex in Buffer, so this is thread safe